synapseclient/__main__.py

```diff
@@ -371,6 +395,42 @@ def getAnnotations(args, syn):
             f.write('\n')
 
 
+def _filter_results_with_path_md5(syn_id, limitSearch, syn):
+    results = syn.restGET('/entity/md5/%s' % utils.md5_for_file(syn_id).hexdigest())['results']
+    if limitSearch is not None:
+        results = filter_id_by_limitSearch(syn, results, limitSearch)
+    set_id_result = _get_unique_ids(results)
+    if len(set_id_result) > 1:
+        raise SynapseError(f"{syn_id} matched more than one file: {set_id_result}")
+    return results
+
+
+def filter_id_by_limitSearch(syn, results, limitSearch):
+    # Go through and find the path of every entity found
+    paths = [syn.restGET('/entity/%s/path' % ent['id']) for ent in results]
+    # Filter out all entities whose path does not contain limitSearch
+    results = [ent for ent, path in zip(results, paths) if utils.is_in_path(limitSearch, path)]
+    return results
+
+
+def _get_unique_ids(results):
+    set_id_result = set()
+    for res in results:
+        set_id_result.add(res['id'])
+    return set_id_result
+
+
+def _get_id_from_args(args, syn, command):
```

Comment:
Can we obtain the command from the argparse so that it is automatically generated from the command and if the command is changed for whatever reason it will automatically be updated? e.g. https://docs.python.org/3/library/argparse.html#prog